Sources/Auth/AuthClient.swift

```diff
@@ -1,31 +1,121 @@
+    public let logger: ConsoleLogging
 
-    init(appPairService: AppPairService, appRequestService: AuthRequestService, walletPairService: WalletPairService) {
+    private var account: Account?
+
+    init(appPairService: AppPairService,
+         appRequestService: AppRequestService,
+         appRespondSubscriber: AppRespondSubscriber,
+         walletPairService: WalletPairService,
+         walletRequestSubscriber: WalletRequestSubscriber,
+         walletRespondService: WalletRespondService,
+         account: Account?,
+         rpcHistory: RPCHistory,
+         cleanupService: CleanupService,
+         logger: ConsoleLogging,
+         pairingStorage: WCPairingStorage) {
         self.appPairService = appPairService
         self.appRequestService = appRequestService
         self.walletPairService = walletPairService
-    }
+        self.walletRequestSubscriber = walletRequestSubscriber
+        self.walletRespondService = walletRespondService
+        self.appRespondSubscriber = appRespondSubscriber
+        self.account = account
+        self.rpcHistory = rpcHistory
+        self.cleanupService = cleanupService
+        self.logger = logger
+        self.pairingStorage = pairingStorage
 
-    func request(params: RequestParams) async throws -> String {
-        let uri = try await appPairService.create()
-        try await appRequestService.request(params: params, topic: uri.topic)
-        return uri.absoluteString
+        setUpPublishers()
     }
 
-    func pair(uri: String) async throws {
+    public func pair(uri: String) async throws {
         guard let pairingURI = WalletConnectURI(string: uri) else {
             throw Errors.malformedPairingURI
         }
         try await walletPairService.pair(pairingURI)
     }
+
+    public func request(_ params: RequestParams, topic: String?) async throws -> String? {
+        logger.debug("Requesting Authentication")
+        if let topic = topic {
+            guard pairingStorage.hasPairing(forTopic: topic) else {
+                throw Errors.noPairingMatchingTopic
+            }
+            logger.debug("Requesting on existing pairing")
+            try await appRequestService.request(params: params, topic: topic)
+            return nil
+        } else {
+            let uri = try await appPairService.create()
+            try await appRequestService.request(params: params, topic: uri.topic)
+            return uri.absoluteString
+        }
+    }
+
+    public func respond(_ params: RespondParams) async throws {
+        guard let account = account else { throw Errors.unknownWalletAddress }
+        try await walletRespondService.respond(respondParams: params, account: account)
+    }
+
+    public func getPendingRequests() throws -> [AuthRequest] {
+        guard let account = account else { throw Errors.unknownWalletAddress }
+        let pendingRequests: [AuthRequest] = rpcHistory.getPending()
+            .filter {$0.request.method == "wc_authRequest"}
```

Comment:
mb move filter and mapping out of AuthClient? rpcHistoryService or pendingService

Reply:
good idea!